abstractAgent.py

```diff
@@ -0,0 +1,10 @@
+class abstractAgent:
```

Comment:
Can we use the decorators described in http://masnun.rocks/2017/04/15/interfaces-in-python-protocols-and-abcs/ if we're using this as an abstract class?